src/main/kotlin/org/rust/ide/notifications/RsUntrustedNotificationProvider.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.notifications
+
+import com.intellij.ide.impl.TrustChangeNotifier
+import com.intellij.ide.impl.UntrustedProjectEditorNotificationPanel
+import com.intellij.ide.impl.isTrusted
+import com.intellij.ide.impl.setTrusted
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.ui.EditorNotificationPanel
+import com.intellij.ui.EditorNotifications
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.cargo.project.model.isCargoToml
+import org.rust.lang.core.psi.isRustFile
+
+class RsUntrustedNotificationProvider : EditorNotifications.Provider<EditorNotificationPanel>(), DumbAware {
+
+    override fun getKey(): Key<EditorNotificationPanel> = PROVIDER_KEY
+
+    @Suppress("UnstableApiUsage")
+    override fun createNotificationPanel(file: VirtualFile, fileEditor: FileEditor, project: Project): EditorNotificationPanel? {
+        if (project.isTrusted()) return null
+        if (!(file.isRustFile || file.isCargoToml)) return null
+
+        val cargoProjects = project.cargoProjects
+        if (!cargoProjects.initialized) return null
+
+        return UntrustedProjectEditorNotificationPanel(project, fileEditor) {
+            project.setTrusted(true)
+            cargoProjects.refreshAllProjects()
```

Comment:
Is it possible to set `setTrusted(true)` from a different place, e.g. from settings? If yes, maybe we should invoke `cargoProjects.refreshAllProjects()` from `TrustChangeNotifier` ?

Reply:
It's not possible from settings (at least, I don't know how) but it's possible from similar things in other build systems (if project is complex, it can contain different build systems), so I've extracted the notifier into a separate file and call `cargoProjects.refreshAllProjects()` from it.
Note, it leads to a double call of `refreshAllProjects` if you mark a project as trusted from dialog shown during invocation of `Reload Cargo Projects` action but I think it's ok since it will be only once